This patch makes shutdown use the Single User runlevel by default so
the gettys are shutdown too.

Roy Marples <uberlord@gentoo.org>

Forward-ported from v2.86 to v2.94_beta
Lars Wendler <polynomial-c@gentoo.org>

--- sysvinit-2.94/man/shutdown.8
+++ sysvinit-2.94/man/shutdown.8
@@ -44,7 +44,7 @@
 \fBshutdown\fP does its job by signalling the \fBinit\fP process, 
 asking it to change the runlevel.
 Runlevel \fB0\fP is used to halt the system, runlevel \fB6\fP is used
-to reboot the system, and runlevel \fB1\fP is used to put to system into
+to reboot the system, and runlevel \fBS\fP is used to put to system into
 a state where administrative tasks can be performed; this is the default
 if neither the \fI-h\fP or \fI-r\fP flag is given to \fBshutdown\fP.
 To see which actions are taken on halt or reboot see the appropriate
--- sysvinit-2.94/src/shutdown.c
+++ sysvinit-2.94/src/shutdown.c
@@ -547,7 +547,7 @@
 		usage();
   		exit(1);
 	}
-	strcpy(down_level, "1");
+	strcpy(down_level, "S");
 	halttype = NULL;
         memset(when, '\0', WHEN_SIZE);
 
@@ -734,6 +734,8 @@
 			strncpy(newstate, "for reboot", STATELEN);
 			break;
 		case '1':
+		case 'S':
+		case 's':
 			strncpy(newstate, "to maintenance mode", STATELEN);
 			break;
 		default:
